### PR TITLE
IFF for allied mobs and unique turrets

### PR DESCRIPTION
**Pull request by Dimasw99**
## About The Pull Request
IFF (Identification Friend or Foe)
This PR allows projectiles to be given the abillity to bypass friendly_to_colony people.
This abillity was then given to the artificer turret and ranged friendly mobs.

Reasoning:
After playing with both the guild turret and the handmade roomba from soteria I noticed lack of practicallity quite quickly.
The guild turrets often or not will friendly fire people by either you standing infront, or behind the enemy and since the turrets shoots in bursts, it will always overkill and shoot you again. Both the roomba suffers from this to a lesser extend but due to its nature, it is always behind you following. Sure, you can move around and manouver it infront of you but it can happen that it will shoot you still even with that 10% chance.
In both cases, if either the guild turret friendly fires or the roomba, you end up with an injury that requires medical to fix. Be it the shrapnel or the organ damage from the roomba.
I think it is also a good point why both features were being so rarely used and those times they were being used, only to be tried out. I believe it would really help robotics feel more useful at making roombas again and guild to help and contribute with making turrets that wont shoot half of blackshield from behind. Do note, this doesn't apply to all turrets. The basic turrets still friendly fire.

Little mention:
Yes the code is quite simple but it works and it can expanded and improved on in the future to contain a proper list of all "friendlies" and not just use friendly_to_colony

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
Can you make it so Exl turrets dont shoot their own too in this pr?
Also exl implants need to toggle colony friendly off so the guild turret *can* friendly fire them
Also Also
"Converted" GP turrets need this set as well

File: code/modules/mob/living/carbon/superior_animal/attack.dm [hunk 1]
```diff
@@ -156,7 +156,7 @@
 					if (possible_target)
 						if (possible_target == target)
 							continue //we made the concious choice to attack them
-						else if (!(prob(do_friendly_fire_chance)) && (((!attack_same && (possible_target.faction == faction)) || (possible_target in friends)) || (possible_target.friendly_to_colony && friendly_to_colony)))
+						else if (!(prob(do_friendly_fire_chance)) && !friendly_to_colony && (((!attack_same && (possible_target.faction == faction)) || (possible_target in friends))))
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/superior_animal/attack.dm:159]: Dosnt this mean that golems (church ones) and friendly mobs dont get shot completely?

Comment (Dimasw99) [reply code/modules/mob/living/carbon/superior_animal/attack.dm:159]: resolved in DM's

**Comment by Dimasw99** [comment]
> Can you make it so Exl turrets dont shoot their own too in this pr? Also exl implants need to toggle colony friendly off so the guild turret _can_ friendly fire them Also Also "Converted" GP turrets need this set as well

I thought about this but for this to be implemented you would need to start creating a list and that is not something I wanted to deal with. Right now the code is limited to just handle colony allies since it is just a simple boolean.

> Can you make it so Exl turrets dont shoot their own too in this pr? Also exl implants need to toggle colony friendly off so the guild turret _can_ friendly fire them Also Also "Converted" GP turrets need this set as well

Done

**Pull request merged**